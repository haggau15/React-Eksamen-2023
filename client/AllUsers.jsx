import * as React from "react";
import {json, useNavigate} from "react-router-dom";
import {useEffect, useState} from "react";

export function ListUsers(){
    const navigate=useNavigate();
    const [users,setUsers]=useState();


    useEffect(()=>  {
        setValues();
        async function setValues(){
           (await fetch("http://localhost:23434/api/user/")).json().then(response =>{
                setUsers((response.temp))
            })}
    },[]);
    if(!users)
    {
        return (<div>loading...</div>)
    }
    async function handleOnclick(e){
        e.preventDefault();
        navigate("/");
    }
    return(
        <div >
            <button onClick={handleOnclick}>Return</button>
            <h1>Users</h1>
            {users.map(u=>
                <div key={u.name}>
                    <h2>{u.name}</h2>
                    <h3>{u.role} ({u.department})</h3>
                </div>)}
        </div>)
}
