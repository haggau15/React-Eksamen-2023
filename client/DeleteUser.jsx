import * as React from "react";
import {useNavigate} from "react-router-dom";
import {useEffect, useState} from "react";

export function DeleteUser() {

    const navigate = useNavigate();
    const [username,setUsername]=useState("");

    async function deleteUser(name) {
        await fetch("/api/user/", {
            method: "DELETE",
            headers: {
                "content-type": "application/json"
            },
            body: JSON.stringify({name})
        });
    }

    async function handleOnclick(e) {
        e.preventDefault();
        navigate("/");
    }
    async function handleSubmit(e) {
        e.preventDefault();
        navigate("/");
        await deleteUser(username);
    }
    return (
        <>
            <button onClick={handleOnclick}>Return</button>
            <h1> Change the values you want to update</h1>
            <form onSubmit={handleSubmit}>
                <div>
                    <label>Username: <input value={username} onChange={e => setUsername(e.target.value)}/></label>
                </div>
                <pre>
                {JSON.stringify({username})}
                </pre>
                <div>
                    <button>Delete</button>
                </div>
            </form>
        </>
    );
}
