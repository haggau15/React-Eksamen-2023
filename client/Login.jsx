import * as React from "react";
import {useNavigate} from "react-router-dom";
import {useState} from "react";



async function onLogin(username, password) {
    await fetch("/api/login/", {
        method: "POST",
        headers: {
            "content-type": "application/json"
        },
        body: JSON.stringify({username, password})
    });
}

export function Login() {
    const navigate=useNavigate();
    const [password,setPassword]=useState("");
    const [username,setUsername]=useState("");

    async function handleSubmit(e)
    {
        e.preventDefault();
        await onLogin(username,password);
        navigate("/");
    }
    async function handleOnclick(e){
        e.preventDefault();
        navigate("/");
    }
    return (<>
            <button onClick={handleOnclick}>Return</button>
            <h1> Log in </h1>
            <form onSubmit={handleSubmit}>
                <div>
                    <label>Username: <input value={username} onChange={e=> setUsername(e.target.value)}/></label>
                </div>
                <div>
                    <label>Password: <input value={password} onChange={e=> setPassword(e.target.value)}/></label>
                </div>
                <pre>
                {JSON.stringify({username,password})}
            </pre>
                <div><button>Login</button></div>
            </form>
        </>
    );
}
export async function Logout()
{
    await fetch("/api/logout/", {
        method: "POST",
        headers: {
            "content-type": "application/json"
        },
    });
}
/**
 * <div>Username: <input type="text" name="username" /></div>
 *                 <div>Password: <input type="password" name="password" /> </div>
 */