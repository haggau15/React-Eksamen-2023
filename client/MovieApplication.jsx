import {useEffect, useState} from "react";
import {useNavigate} from "react-router-dom";
import * as React from "react";

export function ListMovies({moviesAPI}){
    const setValues = async () =>{
        setMovies(undefined);
        setMovies(await moviesAPI.listMovies());
    }
    const [movies,setMovies]=useState();
    useEffect(()=> {
        setValues()
    },[]);
    if(!movies)
    {
        return (<div>loading...</div>)
    }
    return(
        <div >
            <h1>Movies</h1>
            {movies.map(m=>
                <div key={m.title}>
                    <h2>{m.title} ({m.year}</h2>
                    <div> {m.plot}</div>
                </div>)}
        </div>)
}

export function NewMovie({moviesAPI}) {
    const [title,setTitle]=useState("");
    const [plot,setPlot]=useState("");
    const [year,setYear]=useState("");

    const navigate = useNavigate();
    async function handleSubmit(e)
    {
        e.preventDefault();
        await moviesAPI.onAddMovie({plot,title,year});
        navigate("/")
    }
    return(
        <form onSubmit={handleSubmit}>
            <h1>New movie</h1>
            <div>
                <label>Title: <input value={title} onChange={e=> setTitle(e.target.value)}/></label>
            </div>
            <div>
                <label>plot: <input value={plot} onChange={e=> setPlot(e.target.value)}/></label>
            </div>
            <div>
                <label>Title: <input value={year} onChange={e=> setYear(e.target.value)}/></label>
            </div>
            <button>Submit</button>
            <pre>
                {JSON.stringify({title,year,plot})}
            </pre>
        </form>
    )
}