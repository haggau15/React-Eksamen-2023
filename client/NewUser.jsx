import * as React from "react";
import {useNavigate} from "react-router-dom";
import {useEffect, useState} from "react";

async function addUser(username,password,department,role) {
    await fetch("/api/user/", {
        method: "POST",
        headers: {
            "content-type": "application/json"
        },
        body: JSON.stringify({username,password,department,role})
    });
}

export function NewUser()
{
    const navigate=useNavigate();
    const [password,setPassword]=useState("");
    const [username,setUsername]=useState("");
    const [role,setRole]=        useState("");
    const [department,setDepartment]=useState("");

    async function handleSubmit(e)
    {
        e.preventDefault();
        navigate("/");
        await addUser(username,password,department,role);
    }
    async function handleOnclick(e){
        e.preventDefault();
        navigate("/");
    }
    return (<>
            <button onClick={handleOnclick}>Return</button>
            <h1> Enter the values for the new user</h1>
            <form onSubmit={handleSubmit}>
                <div>
                    <label>Username: <input value={username} onChange={e=> setUsername(e.target.value)}/></label>
                </div>
                <div>
                    <label>Password: <input value={password} onChange={e=> setPassword(e.target.value)}/></label>
                </div>
                <div>
                    <label>Department: <input value={department} onChange={e=> setDepartment(e.target.value)}/></label>
                </div>
                <div>
                    <label>Role: <input value={role} onChange={e=> setRole(e.target.value)}/></label>
                </div>
                <pre>
                {JSON.stringify({username,password})}
            </pre>
                <div><button>Update</button></div>
            </form>
        </>
    );
}