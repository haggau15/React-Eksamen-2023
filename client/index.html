<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Task manager</title>
</head>
<body>
<div id="user"></div>
<div id="app"></div>

</body>
<script src="index.jsx" type="module"></script>

<script>
    async function fetchUsername() {
        console.log
        const res = await fetch("/api/login/");
        if(res.ok){
            const {username} = await res.json();
            document.getElementById("user").innerHTML = `<div>Hello ${username}</div>`
        }
    }
    fetchUsername();
</script>

</html>

<!--
<div id="logout"></div>
<div id="login"></div>

            document.getElementById("logout").innerHTML = `<button >Logout</button>`

else {
            document.getElementById("login").innerHTML = "<button onClick={Login}>login</button>"
        }
-->