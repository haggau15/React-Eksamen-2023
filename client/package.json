{
  "dependencies": {
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-router-dom": "^6.8.0"
  },
  "babel": {
    "presets": [
      "@babel/preset-env",
      "@babel/preset-react"
    ]
  },
  "jest": {
    "testEnvironment": "jsdom"
  },
  "name": "reactlesson",
  "description": "* [ ] Say hello to react - parcel generates react app from jsx\r * [ ] Requests in React - Navigate\r * [ ] List existing movies \r * [ ] Add new movie using forms and state\r * [ ] Encapsulate movie database\r * [ ] Express app\r * [ ] Movie API to server",
  "version": "1.0.0",
  "main": "index.js",
  "devDependencies": {
    "@babel/preset-env": "^7.20.2",
    "@babel/preset-react": "^7.18.6",
    "babel-jest": "^29.4.2",
    "jest": "^29.4.2",
    "parcel": "^2.8.3",
    "process": "^0.11.10"
  },
  "scripts": {
    "dev": "../node_modules/.bin/parcel index.html",
    "test": "jest"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/haggau15/ReactMovieDB.git"
  },
  "author": "",
  "license": "ISC",
  "bugs": {
    "url": "https://github.com/haggau15/ReactMovieDB/issues"
  },
  "homepage": "https://github.com/haggau15/ReactMovieDB#readme"
}
