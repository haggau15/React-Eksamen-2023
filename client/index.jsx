import * as React from "react";
import {Route, Link, BrowserRouter, Routes, useNavigate} from "react-router-dom"
import { createRoot } from 'react-dom/client';
import {useEffect, useState} from "react";
import {m} from "caniuse-lite/data/browserVersions";
import {ListMovies, NewMovie} from "./MovieApplication";

function FrontPage() {
return(
     <div>
            <h1>Movie DB</h1>
            <ul>
                <li><Link to="/movies/new">New movie</Link></li>
                <li><Link to="/movies">All movies</Link></li>
            </ul>
        </div>
)
}



function Application(){
    const moviesAPI={
        onAddMovie: async (m) => {
            await fetch("/api/movies",{
                method:"POST",
                headers: {
                    "content-type": "application/json"
                },
                body:JSON.stringify(m)
            })
        },
        listMovies: async () => {
            const res = await fetch("/api/movies");
            return res.json();
        }
    }
        return (
            <BrowserRouter>
                <Routes >
                    <Route path="/" element={<FrontPage/>}/>
                    <Route path="/movies/new" element={<NewMovie moviesAPI={moviesAPI}/>}/>
                    <Route path="/movies" element={<ListMovies moviesAPI={moviesAPI}/>}/>
                </Routes>
            </BrowserRouter>)

}
const container = document.getElementById("app")
const root = createRoot(container);
root.render(<Application/>);
