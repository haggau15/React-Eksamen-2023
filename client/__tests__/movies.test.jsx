import {render} from "react-dom";
import {ListMovies} from "../MovieApplication";
import * as React from "react";
import {createRoot} from "react-dom/client";

describe("movie pages", () => {
    it("Shows movies",()=> {
        const element=document.createElement("div");
        const root = createRoot(element);

        root.render(<ListMovies/>)
        expect(element.innerHTML).toMatchSnapshot();
        console.log(element.innerHTML);

    });
    it("User can add new movie",  ()=> {

    });
});