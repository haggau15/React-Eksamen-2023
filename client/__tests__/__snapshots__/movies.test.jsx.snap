// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movie pages Shows movfies 1`] = `""`;

exports[`movie pages Shows movies 1`] = `""`;
